### PR TITLE
Introduce fine-grained optimiser settings in libevmasm

### PR DESCRIPTION
**Pull request by axic**
Split off #2600.

### CODE REVIEW

File: libevmasm/Assembly.h [hunk 1]
```diff
@@ -97,6 +97,21 @@ class Assembly
...
+	{
+		bool isCreation = false;
+		bool runPeephole = false;
+		bool runDeduplicate = false;
+		bool runCSE = false;
+		bool runConstantOptimiser = false;
+		/// The tradeoff specifies an estimate on how often each opcode in this assembly
+		/// will be executed, i.e. use a small value to optimiser for size and a large value
+		/// to optimise for runtime gas usage.
+		size_t constantOptimiserTradeoff = 200;
```
Comment (axic) [inline libevmasm/Assembly.h]: Let us decide here for a correct naming of this field.

Comment (chriseth) [reply libevmasm/Assembly.h]: expectedExecutionsPerDeployment...

Comment (axic) [reply libevmasm/Assembly.h]: Used this one.

**Comment by axic** [comment]
This should be ready to merge - should have no effect (everything uses the old API)

**Review by chriseth — APPROVED** [review]

**Pull request merged**